armada/utils/git.py

```diff
@@ -3,16 +3,23 @@
 import shutil
 from os import path
 
+from ..exceptions import git_exceptions
+
 def git_clone(repo_url, branch='master'):
     '''
     clones repo to a /tmp/ dir
     '''
 
     if repo_url == '':
+        raise git_exceptions.GitLocationException(repo_url)
         return False
 
     _tmp_dir = tempfile.mkdtemp(prefix='armada', dir='/tmp')
-    pygit2.clone_repository(repo_url, _tmp_dir, checkout_branch=branch)
+
+    try:
+        pygit2.clone_repository(repo_url, _tmp_dir, checkout_branch=branch)
+    except Exception:
```

Comment:
This goes along with my comment above, I will see if pygit has any exceptions that are more specific than GitError.